components/esp_now/esp_now_component.cpp

```diff
@@ -87,9 +87,9 @@ void ESPNowComponent::dump_config() { ESP_LOGCONFIG(TAG, "esp_now:"); }
 #ifdef USE_ESP8266
 void ESPNowComponent::on_data_received(uint8_t *bssid, uint8_t *data, uint8_t len) {
 #elif defined(USE_ESP32)
-void ESPNowComponent::on_data_received(const uint8_t *bssid, const uint8_t *data, int len) {
+void ESPNowComponent::on_data_received(const esp_now_recv_info_t *esp_now_info, const uint8_t *data, int len) {
 #endif
-  auto packet = make_unique<ESPNowPacket>(bssid, data, len);
+  auto packet = make_unique<ESPNowPacket>(esp_now_info->src_addr, data, len);
```

Comment:
This line will need to go inside the ifdef and be copied for esp8266 to remain the same as is now.